addons/stock/static/src/js/lazy_column_list.js

```diff
@@ -0,0 +1,160 @@
+        return dataPoint;
+    },
+});
+
+const LazyColumnListRenderer = ListRenderer.extend({
+    init: function () {
+        this.currentNbLazyRpc = 0;
+        return this._super.apply(this, arguments);
+    },
+
+    _lazyLoad: async function (state, lazyColumns) {
+        const res = await this._rpc(
+            {
+                model: state.model,
+                method: "read",
+                args: [state.res_ids, lazyColumns],
+            },
+            { shadow: true }
+        );
+        const resById = {};
+        for (const element of res.values()) {
+            resById[element.id] = {};
+            for (const lazyCol of lazyColumns) {
+                resById[element.id][lazyCol] = element[lazyCol];
+            }
+        }
+        _bindLazyColumnToState(this.state, lazyColumns, resById);
+        this.currentNbLazyRpc -= 1;
+        this._render(false);
+    },
+
+    async _render(lazy = true) {
+        if (lazy && this.state.res_ids.length > 0) {
+            // TODO don't load if the optional is hidden or all lazy column is invisible
+            this.currentNbLazyRpc += 1;
+        }
+        await this._super(...arguments);
+        if (lazy && this.state.res_ids.length > 0) {
+            const lazyColumns = _getLazyColumns(this.state.fieldsInfo[this.state.type]);
+            this._lazyLoad(this.state, lazyColumns);
+        }
+    },
+
+    _isLazyNode: function (node) {
+        try {
+            if (node.attrs && node.attrs.options && JSON.parse(node.attrs.options).lazy) {
+                return true;
+            }
+        } catch (e) {
+            return false;
+        }
+        return false;
+    },
+
+    _renderBodyCell: function (record, node) {
+        const $res = this._super.apply(this, arguments);
+        const lazyColumns = _getLazyColumns(this.state.fieldsInfo[this.state.type]);
+        if (this._isLazyNode(node)) {
+            // Decorations doesn't work for lazy column because the evalContext isn't updated and it is a readonly attribute
+            // Then a lazy fields can be only used in a other lazy fields decoration.
+            const extraData = {};
+            for (const col of lazyColumns) {
+                extraData[col] = record.data[col];
+            }
+            const evalContext = Object.assign({}, record.evalContext, extraData);
+            for (const [cssClass, expr] of Object.entries(this.fieldDecorations[node.attrs.name])) {
+                $res.toggleClass(cssClass, py.PY_isTrue(py.evaluate(expr, evalContext)));
+            }
+        }
+        return $res;
+    },
+
+    _renderHeaderCell: function (node) {
+        const $res = this._super.apply(this, arguments);
+        if (this._isLazyNode(node)) {
+            if (this.currentNbLazyRpc > 0) {
+                const $spin = $("<i/>")
+                    .addClass("fa")
+                    .addClass("fa-refresh")
+                    .addClass("fa-spin")
+                    .addClass("m-1");
```

Comment:
You can class one time `addClass` with an array:
```suggestion
                const $spin = $("<i/>").addClass(["fa", "fa-refresh", "fa-spin", "m-1"]);
```
... but as you're creating the element, you can also directly set the class:
```javascript
                const $spin = $('<i class="fa fa-refresh fa-spin m-1"/>')
```